spec/unit/face/config_spec.rb

```diff
@@ -112,6 +140,19 @@ def render(action, result)
       subject.set('foo', 'bar', {:section => "baz"})
     end
 
+    it "does not duplicate an existing default section" do
+      contents = <<-CONF
+      [main]
+      myport = 4444
+      CONF
+
+      myfile = StringIO.new(contents)
+      Puppet::FileSystem.stubs(:open).with(path, anything, anything).yields(myfile)
+
+      subject.set('foo', 'bar', {:section => ["main"]})
```

Comment:
Can you add an additional test to verify this behavior when a section isn't specified?